### PR TITLE
physlr_mst: integrate pruning

### PR DESCRIPTION
**Pull request by afshinfard**
implement prunning inside physlr_mst
+ Makefile: physlr mst
+ argument for prunning

### CODE REVIEW

File: physlr/physlr.py [hunk 1]
```diff
@@ -918,27 +918,31 @@ def physlr_degree(self):
...
+                gcomponent = gmst.subgraph(component)
+                if nx.number_of_edges(gcomponent) > 0:
+                    messages = Physlr.determine_reachability_by_message_passing(gcomponent)
+                    gmst_copy = Physlr.prune_branches_of_tree(
+                        gmst_copy, gcomponent, messages, self.args.prune)
+            print(int(timeit.default_timer() - t0),
+                  "Extracted and pruned the MST for branches of size <", self.args.prune, ".",
+                  file=sys.stderr)
+            self.write_graph(gmst_copy, sys.stdout, self.args.graph_format)
+            print(int(timeit.default_timer() - t0), "Wrote the pruned MST.", file=sys.stderr)
```
Comment (sjackman) [inline physlr/physlr.py]: Please extract this code lines 927 to 937 out into a helper function `prune_mst`.

File: physlr/physlr.py [hunk 2]
```diff
@@ -918,27 +936,21 @@ def physlr_degree(self):
...
+        if self.args.prune > 0:
+            gmst = Physlr.prune_mst(gmst, self.args.prune)
+            self.write_graph(gmst, sys.stdout, self.args.graph_format)
+            print(int(timeit.default_timer() - t0), "Wrote the pruned MST.", file=sys.stderr)
+        elif self.args.prune == 0:
+            print(int(timeit.default_timer() - t0), "Extracted the MST.", file=sys.stderr)
+            self.write_graph(gmst, sys.stdout, self.args.graph_format)
+            print(int(timeit.default_timer() - t0), "Wrote the MST.", file=sys.stderr)
+        else:
+            exit("physlr mst: error: wrong parameter --prune. It must be a non-negative integer.")
```
Comment (sjackman) [inline physlr/physlr.py]: Replace 944 to 953 with…
```python
        gmst = nx.algorithms.tree.mst.maximum_spanning_tree(g, weight="n")
        if self.args.prune > 0:
            gmst = Physlr.prune_mst(gmst, self.args.prune)
        self.write_graph(gmst, sys.stdout, self.args.graph_format)
        print(int(timeit.default_timer() - t0), "Wrote the MST.", file=sys.stderr)
```

**Comment by afshinfard** [comment]
changed and rebased. I am waiting for the test to complete.

**Comment by sjackman** [comment]
Merged! Thank you, Amirhossein!

**Comment by afshinfard** [comment]
Thanks Shaun!

**Pull request merged**